.storybook/utils/docscript.js

```diff
@@ -1,70 +1,236 @@
+                entry.properties[name] = {
+                    description: prop.description,
+                    type: prop.type.names[0],
+                    defaultValue: prop.defaultvalue
+                };
+            });
+        }
+
+        if ((doc.kind === 'class' || doc.kind === 'member') && doc.description) {
+            var data = {
+                name: doc.kind === 'class' ? 'constructor': doc.name,
+                description: doc.description,
+            };
+            if (doc.params) {
+                data.params = doc.params.map(param => {
+                    return {
+                        name: param.name,
+                        type: param.type.names[0],
+                        description: param.description
+                    };
+                });
+            }
+
+            entry.methods[data.name] = data;
+        }
+
+        if (doc.kind === 'event' && doc.description) {
+            var data = {
+                name: doc.name,
+                description: doc.description,
+            };
+            if (doc.params) {
+                data.params = doc.params.map(param => {
+                    return {
+                        name: param.name,
+                        type: param.type.names[0],
+                        description: param.description
+                    };
+                });
+            }
+
+            entry.events[data.name] = data;
+        }
+    });
+
+    return entry;
 }
 
+export function getStoryBookDocs(data) {
+    var result = {};
 
-export function getPropertiesForClass(name) {
-    var doc = docs.filter(doc => doc.name === name && (!!doc.properties || !!doc.params));
-    if (doc.length === 0) {
-        return {};
+    // first get parent class properties
+    if (data.parentClasses) {
+        data.parentClasses.forEach(cls => {
+            var parentDocs = getDocsForClass(cls);
+            if (parentDocs) {
+                parentDocs = getStoryBookDocs(parentDocs);
+                result = Object.assign(result, parentDocs);
+            }
+        });
+    }
+
+    // then get interfaces
+    if (data.interfaces) {
+        data.interfaces.forEach(cls => {
+            var parentDocs = getDocsForClass(cls);
+            if (parentDocs) {
+                parentDocs = getStoryBookDocs(parentDocs);
+                result = Object.assign(result, parentDocs);
+            }
+        });
+    }
+
+    // finally get this class
+    // any duplicate definitions will overwrite the previous ones
+    for (var name in data.properties) {
+        var prop = data.properties[name];
+        var typeAndDefault = convertTypeAndValue(prop.type, prop.defaultValue);
+        result[`${name}`] = {
+            name: name,
+            description: prop.description,
+            defaultValue: typeAndDefault[1],
+            control: {
+                type: typeAndDefault[0]
+            },
+            table: {
+                category: 'Properties',
```

Comment:
Would it be possible to also use the subcategory key for inherited properties? (To include the `Inherits: Foobar` headers).

Reply:
I tried various approaches with this before ending up with the current one. I think now that I found a way to do functions and events without using another category for them, we can do what you're suggesting. However I would like to do that a bit later - the current implementation shows all of the properties / functions / events even for inherited classes but just doesn't show which are inherited and which are not.